src/main/scala/com/raphtory/core/config/PulsarController.scala

```diff
@@ -78,36 +78,60 @@ class PulsarController(conf: Config) {
     pulsarAdmin.namespaces().setDeduplicationStatus(namespace, false)
   }
 
-  def createListeningConsumer[T](
+  private def configuredConsumerBuilder[T](
       subscriptionName: String,
-      messageListener: MessageListener[T],
       schema: Schema[T],
-      topics: String*
-  ): Consumer[T] =
+      topics: Seq[String]
+  ): ConsumerBuilder[T] =
     client
       .newConsumer(schema)
       .priorityLevel(0)
       .topics(topics.toList.asJava)
       .subscriptionName(subscriptionName)
-      .subscriptionType(SubscriptionType.Shared)
+      .subscriptionType(SubscriptionType.Exclusive)
```

Comment:
If this is being set below we can probably just remove the line here to be clearer

Reply:
removed